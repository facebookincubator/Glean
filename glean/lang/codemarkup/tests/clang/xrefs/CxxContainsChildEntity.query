query: codemarkup.cxx.CxxContainsChildEntity _
transform: [gensort, []]
