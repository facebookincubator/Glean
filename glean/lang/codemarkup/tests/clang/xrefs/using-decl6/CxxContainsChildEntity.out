[
  "@generated",
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N1" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 9,
                "columnBegin": 1,
                "lineEnd": 11,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "enum_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "Foo" },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N1" } } }
                  }
                }
              },
              "isScoped": false,
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 10,
                "columnBegin": 1,
                "lineEnd": 10,
                "columnEnd": 12
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N2" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 13,
                "columnBegin": 1,
                "lineEnd": 16,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "bar" } } },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N2" } } }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 15,
                "columnBegin": 22,
                "lineEnd": 15,
                "columnEnd": 34
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N2" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 13,
                "columnBegin": 1,
                "lineEnd": 16,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "Bar" },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N2" } } }
                  }
                }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 14,
                "columnBegin": 22,
                "lineEnd": 14,
                "columnEnd": 34
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N3" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 18,
                "columnBegin": 1,
                "lineEnd": 24,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "f" } } },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N3" } } }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 21,
                "columnBegin": 1,
                "lineEnd": 23,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N3" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 18,
                "columnBegin": 1,
                "lineEnd": 24,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "x" },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N3" } } }
                  }
                }
              },
              "type": { "key": "N2::Bar<Foo>" },
              "kind": {
                "global_": { "kind": 0, "attribute": 0, "definition": true }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 20,
                "columnBegin": 1,
                "lineEnd": 20,
                "columnEnd": 14
              }
            }
          }
        }
      }
    }
  }
]